dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6660,13 +6518,45 @@ int processWhere(SELECT_LEX &select_lex,
             break;
 
         ptp = new ParseTree(new LogicOperator("and"));
-        ptp->right(filters);
-        lhs = gwi.ptWorkStack.top();
+        ptp->left(filters);
```

Comment:
Why do you decide to swap branches?

Reply:
Table ordering is reverse in SELECT_LEX::join_list (which is what we use in buildJoin() now) compared to SELECT_LEX::table_list (which we used earlier). So I needed to swap the branches here to keep the filter ordering same as before.